### ISSUE TITLE
[1.12.2] Whitelist/Blacklist for Worldgen in dimensions

### ISSUE THREAD

**Issue by NielsPilgaard** [P3pp3rF1y/AncientWarfare2#555]
Hello!

I'd like to request a config for whitelisting or blacklisting AW2 worldgen in specific dimensions.
AW2 currently generates villages/structures in all dimensions other than -1 and 1 as far as I know, and the structures and NPCs clash with the feel of some dimensions - specifically Advanced Rocketry planets, without oxygen.

I hope you'll consider it :)

**Comment by P3pp3rF1y** [comment]
Sounds reasonable and I am actually currently looking into world gen so depending how easy it is going to be I may even add it as part of those changes.

**Comment by NielsPilgaard** [comment]
Thank you very much!

**Comment by MokahTGS** [comment]
Whitelist is specifically needed for mod packs that add lots of dimensions like advanced rocketry, etc.

**Comment by P3pp3rF1y** [comment]
So now that I got to this I have actually found out that AW already supports whitelist/blacklisting structures on template level (dimensionlist property for the list of dimensions and dimensionwhitelist to set whether this is black or white list). Templates just don't seem to have this defined.

So I have made a small change that defaults these settings to only overworld whitelisted in case template doesn't define where the structure should appear.
